responses.go

```diff
@@ -32,3 +36,53 @@ func (r *SortResponse) Handle(resp imap.Resp) error {
 func (r *SortResponse) WriteTo(w *imap.Writer) error {
 	return errors.New("sortthread: not yet implemented")
 }
+
+func (r *ThreadResponse) Handle(resp imap.Resp) error {
+	name, fields, ok := imap.ParseNamedResp(resp)
+	if !ok || name != "THREAD" {
+		return responses.ErrUnhandled
+	}
+	if threads, err := parseThreadResp(fields); err != nil {
+		return err
+	} else {
+		r.Threads = threads
+	}
+
+	return nil
+}
+
+func parseThreadResp(fields []interface{}) ([]*Thread, error) {
+	var parent *Thread
```

Comment:
I don't think this is going to cut it. We'd need to remember the whole stack of parents to be able to parse correctly

```
(1 2) (3 4)
   ^ parent is set to 2 here
       ^ parent is still 2 here, should be nil
```

I think we need a recursive `parseThread(fields []interface{})` function instead.

Reply:
This method is recursive (line 72) unless I am missing something?

Reply:
Oh, time for a cup of coffee it seems